torch/distributions/distribution.py

```diff
@@ -10,10 +10,15 @@ class Distribution(object):
 
     has_rsample = False
     has_enumerate_support = False
+    support = None
+    params = {}
 
     def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size()):
         self._batch_shape = batch_shape
         self._event_shape = event_shape
+        for param, constraint in self.params.items():
```

Comment:
This should be gated by `if validate_args:`, using a new constructor arg `validate_args=False` that defaults to false.

Reply:
Oh, yes. After I created the PR, I realized that there was a discussion going on in pytorch/pytorch. Sorry I missed it.